internal/policies/apparmor/apparmor.go

```diff
@@ -229,17 +233,88 @@ func (m *Manager) ApplyPolicy(ctx context.Context, objectName string, isComputer
 	if err := os.RemoveAll(oldApparmorPath); err != nil {
 		return fmt.Errorf(i18n.G("can't remove old apparmor directory %q: %v"), oldApparmorPath, err)
 	}
+	return nil
+}
+
+// applyUserPolicy applies apparmor policies for the user object.
+func (m *Manager) applyUserPolicy(ctx context.Context, e entry.Entry, apparmorPath string, username string, assetsDumper AssetsDumper) (err error) {
+	defer decorate.OnError(&err, i18n.G("can't apply user policy"))
+
+	// Create a temporary directory to store the assets
+	tmpdir := filepath.Join(os.TempDir(), fmt.Sprintf("adsys_apparmor_user_%s_%d", username, time.Now().UnixNano()))
+
+	// Dump all assets in order to get our user policy
+	if err := assetsDumper(ctx, "apparmor/", tmpdir, -1, -1); err != nil {
+		return err
+	}
+	defer os.RemoveAll(tmpdir)
+	profilePaths, err := filesFromEntry(e, tmpdir)
+	if err != nil {
+		return err
+	}
+
+	// The user policy is always a single file
+	if len(profilePaths) != 1 {
+		return fmt.Errorf(i18n.G("expected exactly one profile, got %d"), len(profilePaths))
+	}
+	profilePath := profilePaths[0]
+	profileContents, err := os.ReadFile(profilePath)
+	if err != nil {
+		return err
+	}
+	parsedProfile := strings.ReplaceAll(string(profileContents), "{{ADSYS_USER}}", username)
+
+	// Write the profile to the user's apparmor directory, getting the previous
+	// contents if available
+	oldContent, changed, err := writeIfChanged(filepath.Join(apparmorPath, username), parsedProfile)
+	if err != nil {
+		return err
+	}
+	if !changed {
+		return nil
+	}
+
+	if os.Getenv("ADSYS_SKIP_ROOT_CALLS") != "" {
+		return nil
+	}
+
+	// Reload machine policies to ensure that updates to the user policy are applied
```

Comment:
I would avoid the term "policies" here, or give more context: when you wrote the PR description, I almost had a heart attack :p I thought that policy was the AD policies in that context and that we were rerunning the machine thingy :p